rest/config.go

```diff
@@ -532,6 +532,39 @@ func (dbConfig *DbConfig) validatePersistentDbConfig() (errorMessages error) {
 	return multiError.ErrorOrNil()
 }
 
+// validateConfigUpdate combines the results of validate and validateChanges.
+func (dbConfig *DbConfig) validateConfigUpdate(ctx context.Context, old *db.DatabaseContext, validateOIDCConfig bool) error {
+	err := dbConfig.validate(ctx, validateOIDCConfig)
+	var multiErr *base.MultiError
+	if !errors.As(err, &multiErr) {
+		multiErr = multiErr.Append(err)
+	}
+	multiErr = multiErr.Append(dbConfig.validateChanges(ctx, old))
+	return multiErr.ErrorOrNil()
+}
+
+// validateChanges compares the current DbConfig with the "old" config, and returns an error if any disallowed changes
+// are attempted.
+func (dbConfig *DbConfig) validateChanges(ctx context.Context, old *db.DatabaseContext) error {
```

Comment:
I'm not sure about this accepting a `*DatabaseContext` - ideally it'd keep around a reference to the old `DbConfig` but that may require a more extensive refactor.